### ISSUE TITLE
Trouble building

### ISSUE THREAD

**Issue by keith-hall** [thesquash/sgm#1]
Hi,

Thanks for your hard work on this project! I can't wait to try it out and get mnemonics back! Unfortunately, I'm having some trouble building it, maybe you can help me please?

(It's my first time using meson, I just installed it (on Linux Mint) using `sudo apt-get install meson` which also installed the `ninja-build` package.)

```
❯ meson ..
The Meson build system
Version: 0.53.2
Source dir: /home/keith/repos/sgm
Build dir: /home/keith/repos/sgm/.build
Build type: native build
Project name: sgm
Project version: 0.90.0
C compiler for the host machine: cc (gcc 9.3.0 "cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0")
C linker for the host machine: cc ld.bfd 2.34
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Run-time dependency gtk+-3.0 found: YES 3.24.20
meson.build:10: WARNING: Passed invalid keyword argument "internal".
WARNING: This will become a hard error in the future.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/mesonbuild/mesonmain.py", line 129, in run
    return options.run_func(options)
  File "/usr/lib/python3/dist-packages/mesonbuild/msetup.py", line 245, in run
    app.generate()
  File "/usr/lib/python3/dist-packages/mesonbuild/msetup.py", line 159, in generate
    self._generate(env)
  File "/usr/lib/python3/dist-packages/mesonbuild/msetup.py", line 192, in _generate
    intr.run()
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreter.py", line 4167, in run
    super().run()
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 412, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
    raise e
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 443, in evaluate_statement
    return self.assignment(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 1064, in assignment
    value = self.evaluate_statement(node.value)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 441, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 779, in function_call
    (posargs, kwargs) = self.reduce_arguments(node.args)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 1033, in reduce_arguments
    reduced_pos = [self.evaluate_statement(arg) for arg in args.arguments]
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 1033, in <listcomp>
    reduced_pos = [self.evaluate_statement(arg) for arg in args.arguments]
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 445, in evaluate_statement
    return self.method_call(cur)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 828, in method_call
    return obj.method_call(method_name, args, kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 325, in method_call
    return method(args, kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 231, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 125, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 174, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3/dist-packages/mesonbuild/interpreter.py", line 488, in variable_method
    return self.held_object.get_variable(**kwargs)
TypeError: get_variable() got an unexpected keyword argument 'internal'

```

**Comment by thesquash** [comment]
I used a deprecated keyword in hopes to allow Debian users to compile the package, but it looks like Debian has a new enough version of Meson anyway.  I just updated the build file.  Please pull the new commit and try building again.

**Comment by keith-hall** [comment]
Thanks, it worked perfectly! :)

Am I right in thinking that my `~/.config/gtk-3.0/settings.ini` file should look like this:

```
[Settings]
    gtk-modules = [libsgm-flexible-mnemonics.so]
    gtk-enable-mnemonics = 1
    gtk-auto-mnemonics = 0
```

Because at the moment I'm not seeing mnemonics in the menu bar in Gtk applications... I tried restarting my computer in case it would help. Where can I look for logs to help debug this? :) Sorry for the noobish questions!

**Comment by thesquash** [comment]
Sorry, no, the `settings.ini` file should look like this:

```
[Settings]
gtk-modules = sgm-flexible-mnemonics
gtk-enable-mnemonics = 1
gtk-auto-mnemonics = 0
```

I should probably include an `INSTALL` file for installation and usage instructions. I'll write one up.

**Comment by keith-hall** [comment]
Thanks for the clarification. I tried that and still didn't see any mnemonics.

Then I tried explicitly setting the GTK3_MODULES env var before launching a GTK application and it worked!

```bash
GTK_DEBUG=interactive G_MESSAGES_DEBUG=all GTK3_MODULES=sgm-flexible-mnemonics thunar
```
![image](https://user-images.githubusercontent.com/11882719/115124946-de341780-9fcd-11eb-87f7-864c10340a3d.png)

so I'm not sure why the `settings.ini` route isn't working for me, but I believe I can try setting the `GTK3_MODULES` environment variable in one of my startup scripts to solve my problem.

Thanks once again for your time helping me resolve this, and working on/sharing this project. I'll let you decide whether to keep this issue open or not as a reminder for including an `INSTALL` file :)
